clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1420,18 +1407,20 @@ Sema::DeviceDiagBuilder Sema::SYCLDiagIfDeviceCode(SourceLocation Loc,
                                                    unsigned DiagID) {
   assert(getLangOpts().SYCLIsDevice &&
          "Should only be called during SYCL compilation");
-  DeviceDiagBuilder::Kind DiagKind = [this] {
-    if (isKnownEmitted(*this, dyn_cast<FunctionDecl>(CurContext)))
+  FunctionDecl *FD = dyn_cast<FunctionDecl>(CurContext);
```

Comment:
Shouldn't it be CurLexicalContext?

Reply:
I believe it SHOULD be getCurLexicalContext, but the current implementation was CurContext.  I couldn't come up with a reproducer that I was able to justify changing the status quo here.

Reply:
I should double check, because it was before long vacation, but in my patch I've implemented a new usage of deferred diagnostics - diagnosing of `__float128` usage in device code (https://github.com/intel/llvm/pull/971/files#diff-643c0b5e3365e57856a6871266a81b35) and it didn't work before I changed it to `getCurLexicalContext` .

Reply:
Neat!  Thanks for providing justification.   I'll upload an updated patch momentarily.